test/extensions/filters/http/health_check/health_check_test.cc

```diff
@@ -409,6 +414,104 @@ TEST_F(HealthCheckFilterCachingTest, NotHcRequest) {
             filter_->decodeHeaders(request_headers_no_hc_, true));
 }
 
+TEST_F(HealthCheckFilterNoPassThroughTest, HealthCheckStats) {
+  // Test that health check request counter increases
+  EXPECT_EQ(0, stats_store_.counter("test.health_check.request_total").value());
+  EXPECT_EQ(Http::FilterHeadersStatus::StopIteration,
+            filter_->decodeHeaders(request_headers_, true));
+  EXPECT_EQ(1, stats_store_.counter("test.health_check.request_total").value());
+
+  // Test failed health check stats
+  EXPECT_CALL(context_, healthCheckFailed()).WillOnce(Return(true));
+  Http::TestResponseHeaderMapImpl failed_response{{":status", "503"}};
+  EXPECT_CALL(callbacks_, encodeHeaders_(HeaderMapEqualRef(&failed_response), true));
+  filter_->decodeHeaders(request_headers_, true);
+  EXPECT_EQ(1, stats_store_.counter("test.health_check.failed").value());
+
+  // Test healthy response stats
+  EXPECT_CALL(context_, healthCheckFailed()).WillOnce(Return(false));
+  Http::TestResponseHeaderMapImpl healthy_response{{":status", "200"}};
+  EXPECT_CALL(callbacks_, encodeHeaders_(HeaderMapEqualRef(&healthy_response), true));
+  filter_->decodeHeaders(request_headers_, true);
+  EXPECT_EQ(2, stats_store_.counter("test.health_check.ok").value());
```

Comment:
I'm assuming that prior to this test-block the value of `test.health_check.ok` was 1. Please add an EXPECT_EQ at the beginning of the block, or add the expectation to the block above that increases the value.